### DISCUSSION TITLE
[question] nanomq for increasing number of users. "user-password" couple adding while broker launched

### DISCUSSION THREAD

**Discussion by luxor092** [nanomq/nanomq#1866] [Q&A] [answered]
Good day.

 I am just started with mqtt and nanomq in particular, the question may be a rookie one, but I don't know yet how to resolve. I try to make a scalable system: There are increasing number of customers. Each customer buys one hardware IoT device. After purchasing a device, the customer does a quick registration on the internet, which creates "user-password" couple for the device to use.  But I found I cannot modify **nanomq_pwd.conf** file while nanomq broker is working. (Or can I?) So, as I assume, I have to restart broker every time when new customer makes a registration. That will lead to connection loss for every other customers who are connected to the broker at the moment. My question is: what can I do to handle it the best way? Another question: is there any limitation about number of user-password couples in the nanomq_pwd.conf? And a little bit out of topic, may be there are some other **common methods** to handle increasing users without adding user-password pair at all?
Thank you in advance!

**Comment by JaylinYu** [comment, marked as answer]
Basically, you need a hot update on nanomq_pwd configurations. Unfortunately, this is not supported for now. It has been on our roadmap for a long time, but there is just not enough time for it.

**is there any limitation about number of user-password couples in the nanomq_pwd.conf?**
No, there isn't, but an extremely long user-pwd pair list will reduce the connecting performance.

**may be there are some other common methods to handle increasing users without adding user-password pair at all?**
ACL is similar, or HTTP authentication is the best way out for you now.
https://nanomq.io/docs/en/latest/config-description/acl.html#http-authentication
You can export CONNECT to your own HTTP server, then you can freely doing your own authentication rules there.

**Reply by luxor092** [reply]
I have read and understood.
Thank you for providing the answers!
